File: agent/app/model/dns.go
package model

type DnsZone struct {
	BaseModel
	Name       string `gorm:"not null;uniqueIndex" json:"name"`
	Type       string `gorm:"not null;default:'Native'" json:"type"`
	MasterIP   string `json:"masterIP"`
	SoaPrimary string `json:"soaPrimary"`
	SoaEmail   string `json:"soaEmail"`
	SoaRefresh int    `gorm:"default:10800" json:"soaRefresh"`
	SoaRetry   int    `gorm:"default:3600" json:"soaRetry"`
	SoaExpire  int    `gorm:"default:604800" json:"soaExpire"`
	SoaTTL     int    `gorm:"default:3600" json:"soaTTL"`
	DefaultTTL int    `gorm:"default:3600" json:"defaultTTL"`
	Status     string `gorm:"not null;default:'active'" json:"status"`
	PdnsID     string `json:"pdnsID"`
}

type DnsRecord struct {
	BaseModel
	ZoneID   uint   `gorm:"not null;index" json:"zoneID"`
	Name     string `gorm:"not null" json:"name"`
	Type     string `gorm:"not null" json:"type"`
	Content  string `gorm:"not null" json:"content"`
	TTL      int    `gorm:"default:3600" json:"ttl"`
	Priority int    `json:"priority"`
	Disabled bool   `gorm:"default:false" json:"disabled"`
	AutoGen  bool   `gorm:"default:false" json:"autoGen"`
}

File: frontend/src/api/interface/dns.ts
import { CommonModel, ReqPage } from '.';

export namespace Dns {
    export interface Zone extends CommonModel {
        name: string;
        type: string;
        masterIP: string;
        soaPrimary: string;
        soaEmail: string;
        soaRefresh: number;
        soaRetry: number;
        soaExpire: number;
        soaTTL: number;
        defaultTTL: number;
        status: string;
        pdnsID: string;
    }

    export interface ZoneRes extends Zone {
        recordCount: number;
    }

    export interface ZoneSearch extends ReqPage {
        info: string;
        orderBy: string;
        order: string;
    }

    export interface ZoneCreate {
        name: string;
        type: string;
        masterIP?: string;
        soaPrimary?: string;
        soaEmail: string;
        defaultTTL?: number;
    }

    export interface ZoneUpdate {
        id: number;
        soaPrimary?: string;
        soaEmail?: string;
        soaRefresh?: number;
        soaRetry?: number;
        soaExpire?: number;
        soaTTL?: number;
        defaultTTL?: number;
    }

    export interface Record extends CommonModel {
        zoneID: number;
        name: string;
        type: string;
        content: string;
        ttl: number;
        priority: number;
        disabled: boolean;
        autoGen: boolean;
    }

    export interface RecordRes extends Record {
        zoneName: string;
    }

    export interface RecordSearch extends ReqPage {
        zoneID: number;
        type?: string;
        info?: string;
    }

    export interface RecordCreate {
        zoneID: number;
        name: string;
        type: string;
        content: string;
        ttl?: number;
        priority?: number;
    }

    export interface RecordUpdate {
        id: number;
        name: string;
        type: string;
        content: string;
        ttl?: number;
        priority?: number;
        disabled?: boolean;
    }

    export interface Status {
        enabled: boolean;
        containerUp: boolean;
        containerName: string;
        version: string;
    }
}
